package/MDAnalysis/coordinates/timestep.pyx

```diff
@@ -0,0 +1,982 @@
+        self.has_forces = True
+        if cnp.PyArray_Check(new_forces):  # is it an array?
+            cnp.PyArray_CopyInto(self._forces, new_forces)
+            # copy into target, handles dtype conversion
+            # should we check return code?
+        else:
+            self._forces[:] = new_forces
+
+    @classmethod
+    def from_timestep(cls, Timestep other, **kwargs):
+        """Create a copy of another Timestep, in the format of this Timestep
+
+        .. versionadded:: 0.11.0
+        """
+        ts = cls(other.n_atoms,
+                 positions=other.has_positions,
+                 velocities=other.has_velocities,
+                 forces=other.has_forces,
+                 **kwargs)
+        ts.frame = other.frame
+        if other.dimensions is not None:
+            ts.dimensions = other.dimensions.copy(order=cls.order)
+        try:
+            ts.positions = other.positions.copy(order=cls.order)
+        except NoDataError:
+            pass
+        try:
+            ts.velocities = other.velocities.copy(order=cls.order)
+        except NoDataError:
+            pass
+        try:
+            ts.forces = other.forces.copy(order=cls.order)
+        except NoDataError:
+            pass
+
+
+        try:
+            other._reader = weakref.ref(ts._reader())
+        except TypeError:  # TypeError from calling None weakref
+            pass
+
+        ts.data = copy.deepcopy(other.data)
+
+        return ts
+
+    @classmethod
+    def from_coordinates(cls,
+                         positions=None,
+                         velocities=None,
+                         forces=None,
+                         **kwargs):
+        """Create an instance of this Timestep, from coordinate data
+
+        Can pass position, velocity and force data to form a Timestep.
+
+        .. versionadded:: 0.11.0
+        """
+        has_positions = positions is not None
+        has_velocities = velocities is not None
+        has_forces = forces is not None
+
+        lens = [len(a) for a in [positions, velocities, forces]
+                if a is not None]
+        if not lens:
+            raise ValueError("Must specify at least one set of data")
+        n_atoms = max(lens)
+        # Check arrays are matched length?
+        if not all(val == n_atoms for val in lens):
+            raise ValueError("Lengths of input data mismatched")
+
+        ts = cls(n_atoms,
+                 positions=has_positions,
+                 velocities=has_velocities,
+                 forces=has_forces,
+                 **kwargs)
+        if has_positions:
+            ts.positions = positions
+        if has_velocities:
+            ts.velocities = velocities
+        if has_forces:
+            ts.forces = forces
+
+        return ts
+
+    def __eq__(self, other):
+        """Compare with another Timestep
+
+        .. versionadded:: 0.11.0
+        """
+        if not isinstance(other, Timestep):
+            return False
```

Comment:
I know this copies previous behaviour, but I think I've since learnt that the appropriate response here is to return `return NotImplemented` which iirc then checks if `other.__eq__(self)` returns anything better.  This is how an `other.__class__` could say it is equal to this, despite this not knowing of other.